dask/dataframe/io/io.py

```diff
@@ -153,6 +153,16 @@ def from_array(x, chunksize=50000, columns=None, meta=None):
     return new_dd_object(dsk, name, meta, divisions)
 
 
+@overload
+def from_pandas(data: pd.DataFrame) -> DataFrame:
```

Comment:
It looks like `mypy` isn't completely happy with these `overloads` yet. Not a type checking expert, but looking at other places we're using `overload`, we're specifying the other parameters like `npartitions`, `chunksize`, etc

Reply:
Yes, the overloads need to be consistent with the function signature. However, I took a look at this and am seeing a possible issue with the pandas type stubs, so there may be a more fundamental problem.

Reply:
Thanks for chiming in all.

@ian-r-rose it worked for you?  Even if I match the function signature I get the mypy error: `error: Overloaded function signature 2 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]`.  Same error even if I swap the order

Reply:
Ah, I see what's happening. Yes, it worked for me, but only when I installed the pandas type stubs (`pip install pandas-stubs`). These are not currently in the `pre-commit` environment, cf: https://github.com/dask/dask/issues/9220

Until we can add those stubs, I think it would be okay to add a `type: ignore` with a link to that issue on the relevant line.